numba/tests/test_jitclasses.py

```diff
@@ -1154,6 +1154,18 @@ def test_jitclass_isinstance(obj):
         self.assertEqual(pyfunc(Bar(123)), cfunc(Bar(123)))
         self.assertEqual(pyfunc(0), cfunc(0))
 
+    def test_jitclass_unsupported_dunder(self):
+        with self.assertRaises(TypeError) as e:
+            @jitclass
+            class Foo(object):
+                def __init__(self):
+                    return
+
+                def __enter__(self):
+                    return None
+            Foo()
+        self.assertIn("__enter__", str(e.exception))
```

Comment:
```suggestion
        self.assertIn("Method '__enter__' is not supported.", str(e.exception))
```
I think being more specific here would be beneficial so as to ensure the exception definitely came from the expected location?